controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -114,6 +118,14 @@ func (a *Adapter) EnsureStatusReportedInSnapshot() (controller.OperationResult,
 		a.logger.Error(err, "Failed to update pipeline status in snapshot")
 		return controller.RequeueWithError(fmt.Errorf("failed to update test status in snapshot: %w", err))
 	}
+
+	if !gitops.IsSnapshotCreatedByPACPullRequestEvent(snapshot) && (h.HasPipelineRunFinished(a.pipelineRun) || pipelinerunStatus == intgteststat.IntegrationTestStatusDeleted) {
```

Comment:
```suggestion
	if !gitops.IsSnapshotCreatedByPACPullRequestEvent(snapshot) && ((h.HasPipelineRunFinished(a.pipelineRun) || pipelinerunStatus == intgteststat.IntegrationTestStatusDeleted)) {
```

It should be like this right? Only non pull request snapshots.

Also it may deserve a comment why we care only about PAC snapshots here and where are the other cases handled

Reply:
or is it done in such way to handle tests deletion also for PR integration tests?

Reply:
I don't think adding those round brackets would make any difference since it's already inside a round bracket, right?
This part of the code would only handle non-PR Intg PLRs. I've added a comment to address your concern.

Reply:
if we care only only about PAC runs comment won't save it, it is time to learn about precendce of operators in go.

https://www.tutorialspoint.com/go/go_operators_precedence.htm

```
func main() {
    //                #isPR #finished # deleted                 
	fmt.Println("%v", true && true || true)
	fmt.Println("%v", false && true || true)
	fmt.Println("%v", true && (true || true))
	fmt.Println("%v", false && (true || true))
}
```
https://goplay.tools/snippet/IUawbPTWH4a

Reply:
ah nevermind, I'm totally blind and stupid :D

Reply:
I really appreciate the time you take to explain things to me